src/System.Drawing.Primitives/src/System/Drawing/Color.cs

```diff
@@ -645,9 +645,8 @@ public override bool Equals(object obj)
 
         public override int GetHashCode()
         {
-            return unchecked(value.GetHashCode() ^
-                    state.GetHashCode() ^
-                    knownColor.GetHashCode());
+            int hash = value.GetHashCode();
+            return unchecked((hash << 5) - hash + (knownColor.GetHashCode() << 3) + state);
```

Comment:
@JonHanna, no. I was unable, however, to find any evidence online that 31 is better than 33 just because it's prime, so I just went with something that was similar to what `Tuple` already had. I'll write a test app for this when I get home, but I suspect that the difference in collisions, if any, will be minimal.